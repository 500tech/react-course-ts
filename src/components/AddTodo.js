import React, { Component, createRef } from 'react';

export class AddTodo extends Component {
  state = {
    draft: '',
  };

  componentDidMount() {
    this._input.current.focus();
  }

  _input = createRef();

  onDraftChange = event => {
    const { target } = event;
    const { value } = target;
    this.setState({ draft: value });
  };

  addTodo = () => {
    this.props.onAddTodo(this.state.draft);
    this.setState({ draft: '' });
  };

  render() {
    return (
      <div>
        <input
          ref={this._input}
          value={this.state.draft}
          onChange={this.onDraftChange}
        />
        <button disabled={this.state.draft === ''} onClick={this.addTodo}>
          Add Me!
        </button>
      </div>
    );
  }
}
