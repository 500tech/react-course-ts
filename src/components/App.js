import React, { useState } from 'react';
import styled from 'styled-components';
import uuid from 'uuid';
import produce from 'immer';
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';

import { Greeting } from './Greeting';
import { Todos } from '../pages/Todos';
import { Home } from '../pages/Home';

const Container = styled.div`
  border: 1px solid lightblue;
`;

const INITIAL_STATE = [
  { $id: uuid(), text: 'This is todo #1', done: true },
  { $id: uuid(), text: 'This is todo #2', done: false },
  { $id: uuid(), text: 'This is todo #3', done: false },
  { $id: uuid(), text: 'This is todo #4', done: false },
  { $id: uuid(), text: 'This is todo #5', done: false },
];

const getTodo = text => ({ text, done: false, $id: uuid() });

export function App() {
  const [todos, setTodos] = useState(INITIAL_STATE);
  const toggleTodo = todoIndex =>
    setTodos(
      produce(todos, draftState => {
        draftState[todoIndex].done = !draftState[todoIndex].done;
      })
    );
  const onAddTodo = text =>
    setTodos(
      produce(todos, draftState => {
        draftState.push(getTodo(text));
      })
    );
  return (
    <Router>
      <Container>
        <Greeting name="foobar" />
        <Switch>
          <Route path="/" exact component={Home} />
          <Route
            path="/todos"
            render={() => (
              <Todos
                todos={todos}
                onAddTodo={onAddTodo}
                toggleTodo={toggleTodo}
              />
            )}
          />
        </Switch>
      </Container>
    </Router>
  );
}
