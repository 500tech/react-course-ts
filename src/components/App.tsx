import React, { useState } from "react";
import uuid from "uuid";
import { ThemeProvider } from "styled-components";
import { TodoAdder2 } from "./TodoAdder";
import { TodoList } from "./TodoList";
import { useTodosService } from "../services/todos";
import { lightTheme, darkTheme } from "./theme";

function NoItemsEmptyState() {
  return <p>Oh noes, no items yet! Please create one :)</p>;
}

const App: React.FC = () => {
  const { todos, toggleTodo, removeTodo, addTodo } = useTodosService([
    { id: uuid(), text: "Learn hooks", done: false },
    { id: uuid(), text: "Eat breakfast", done: true }
  ]);
  const [theme, setTheme] = useState<typeof lightTheme | typeof darkTheme>(
    lightTheme
  );

  return (
    <ThemeProvider theme={theme}>
      <div className="container">
        <h1
          onClick={() =>
            setTheme(theme === lightTheme ? darkTheme : lightTheme)
          }
        >
          Todo list
        </h1>
        <TodoAdder2 onAddTodo={addTodo} />
        {todos.length ? (
          <TodoList
            items={todos}
            onToggleTodo={toggleTodo}
            onRemoveTodo={removeTodo}
          />
        ) : (
          <NoItemsEmptyState />
        )}
      </div>
    </ThemeProvider>
  );
};

export default App;
