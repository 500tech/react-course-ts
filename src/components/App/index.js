import { connect } from 'react-redux';
import BaseApp from './App';
import * as countActions from '../../state/actions/count.actions';

function mapStateToProps(state) {
  return {
    count: state.count,
  };
}

const betterMapDispatchToProps = {
  incrementCount: countActions.increment,
  decrementCount: countActions.decrement,
  setCount: countActions.setCount,
};

function mockMapDispatchToProps(dispatch) {
  return Object.keys(betterMapDispatchToProps).reduce((mapper, prop) => {
    mapper[prop] = (...args) => dispatch(betterMapDispatchToProps[prop](...args));
    return mapper;
  }, {});
}

function mapDispatchToProps(dispatch) {
  return {
    incrementCount() {
      dispatch(countActions.increment());
    },
    decrementCount() {
      dispatch(countActions.decrement());
    },
    setCount(count) {
      dispatch(countActions.setCount(count));
    },
  };
}

const connector = connect(
  mapStateToProps,
  mockMapDispatchToProps
);

export const App = connector(BaseApp);
