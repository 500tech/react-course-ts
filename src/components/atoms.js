import styled from 'styled-components';

export const Button = styled.button.attrs({ type: 'button' })`
  &:disabled {
    background-color: transparent;
    border: none;
  }
`;

export const PrimaryButton = styled(Button)`
  background-color: ${props => props.theme.palette.primary};
  color: ${props => props.theme.palette.primaryText};
`;

export const Input = styled.input.attrs({ type: 'text' })`
  &:active {
    outline-color: green;
  }
`;
