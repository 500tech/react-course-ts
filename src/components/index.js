export { App } from './App';
export { Counter } from './Counter';
