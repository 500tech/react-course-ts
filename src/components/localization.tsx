import * as React from 'react';
import { FormattedMessage, injectIntl } from 'react-intl';

export const TranslateAsChild = () => (
    <FormattedMessage id="fake.id" />
);

export const TranslateAsProp = injectIntl(({ intl }) => (
    <div data-test-id={ intl.formatMessage({ id:'fake.id' }) }>Hello world</div>
));
