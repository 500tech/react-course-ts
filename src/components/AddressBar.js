import React, { useState, useEffect } from 'react';
import styled from 'styled-components';
import { withRouter } from 'react-router-dom';
import { Button } from './atoms';

const HorizontalForm = styled.form`
  display: flex;
`;

function useDerivedState(value) {
  const [state, setState] = useState(value);
  useEffect(() => {
    setState(value);
  }, [value]);
  return [state, setState];
}

function BaseAdresssBar({ history, location }) {
  const [url, setUrl] = useDerivedState(location.pathname);
  const go = e => {
    e.preventDefault();
    history.push(url);
  };
  return (
    <HorizontalForm onSubmit={go}>
      <Button onClick={history.goBack}>Back</Button>
      <input value={url} onChange={e => setUrl(e.target.value)} />
      <Button onClick={history.goForward}>Forward</Button>
    </HorizontalForm>
  );
}

export const AddressBar = withRouter(BaseAdresssBar);
