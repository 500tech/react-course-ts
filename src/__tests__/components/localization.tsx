import * as React from 'react';
import { TranslateAsChild, TranslateAsProp } from "../../components/localization";
import {mount} from "enzyme";

jest.mock('react-intl');

describe('<TranslateAsChild/>', () => {
  test('should render', () => {
    const component = mount(
        <TranslateAsChild />
    );

    expect(component).toMatchSnapshot();
  });
});

describe('<TranslateAsProp', () => {
  test('should render', () => {
    const component = mount(
        <TranslateAsProp />
    );

    expect(component.find('div')).toHaveProp('data-test-id', 'fake.id');
  })
});


