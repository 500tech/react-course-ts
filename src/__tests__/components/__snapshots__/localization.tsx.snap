// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TranslateAsChild/> should render 1`] = `
<Component>
  <Component
    id="fake.id"
  >
    <span>
      fake.id
    </span>
  </Component>
</Component>
`;
